internal/services/formatters/csharp/dotnet_cli/formatter.go

```diff
@@ -159,6 +156,19 @@ func (f *Formatter) newVulnerability(dependency *dotnetDependency, projectSubPat
 	return f.SetCommitAuthor(vulnHash.Bind(vuln))
 }
 
+func (f *Formatter) getCodeFilePathAndLine(dependency *dotnetDependency,
+	projectSubPath string) (string, string, string,
+) {
+	code, filePath, line, err := file.GetDependencyCodeFilepathAndLine(
+		f.GetConfigProjectPath(), projectSubPath, dependency.Name, CsProjExt,
+	)
+	if err != nil {
+		f.SetAnalysisError(err, tools.DotnetCli, err.Error(), "")
+		logger.LogErrorWithLevel(messages.MsgErrorGetDependencyCodeFilepathAndLine, err)
+	}
```

Comment:
We should continue even some error occur? I don't think that could be a good approach. If an error is returned the code, filePath and line will be empty, which can cause a lot of confusing, because the output shows an error but the vulnerability is still present. So I think that we should set the analysis error if exists or add the vulnerability if no error occurred.

Reply:
the approach we were using was just ignoring and not displaying the error, this at least notify us if some error occurred.  The error handling we're using is too confuse in my opinion(if something occurs just notify at the end of the analysis), i believe this is the best approach we could use without make some big change on the error handling case.

Reply:
I don't think that would be a big change anyway, we just need to propagate the error to parent calls until `StartAnalysis`.

I agree with you that the our error handling is too confuse, on the entire project btw.

Reply:
but `StartAnalysis` does not return an error, we'll be just setting the analysis error on another place, just moving the same problem to another place

Reply:
We could change the  loop on [`parseOutput`](https://github.com/ZupIT/horusec/blob/90dfac36098bf06c0ac4aa6fbaf04c987e240e2b/internal/services/formatters/csharp/dotnet_cli/formatter.go#L83) to something like this:

```go
for _, value := range strings.Split(output[startIndex:], separator) {
	dependency := f.parseDependencyValue(value)
	if dependency != nil && *dependency != (dotnetDependency{}) {
		vuln, err := f.newVulnerability(dependency, projectSubPath)
		if err != nil {
			f.SetAnalysisError(...)
			continue
		}
		f.AddNewVulnerabilityIntoAnalysis(vuln)
	}
}
```
With this we add the error on analysis results and add just correctly vulnerabilities on analysis result. 


My main point here is that I think we should not add incomplete vulnerabilities data on the analysis result.

Reply:
i understand, i just don't think this PR should be the one we do it, we'll have to change all the formatters to this pattern, this dotnet will be the only one who does not reports incomplete vulnerabilities(and i think reporting incomplete vulnerabilities is still debatable).

Reply:
I don't think that we should change for all formatters because not all formatters can generate an error to fetch the code (only the formatters that was changed on this pr would be effected). On this PR is already changing these formatters to handle the errors, I don't know why we shouldn't improve this here.

I will resolve this conversation for further discussion, but I disagre with the approach used here at all.

Reply:
I agree with @matheusalcantarazup I don't see a problem with making this improvement here only in the formatters changed here

Reply:
I agree with @matheusalcantarazup I don't see a problem with making this improvement here only in the formatters changed here

Reply:
I also agree with Matheus idea of ​​not adding incomplete vulnerabilities. And since we are changing the error handling code, why not make this improvement